src/System.Text.Json/tests/Serialization/OptionsTests.cs

```diff
@@ -8,6 +8,30 @@ namespace System.Text.Json.Serialization.Tests
 {
     public static partial class OptionsTests
     {
+        [Fact]
+        public static void SetOptionsFail()
+        {
+            var options = new JsonSerializerOptions();
+
+            JsonSerializer.Parse<int>("1", options);
+
+            // Verify defaults and ensure getters do not throw.
+            Assert.False(options.AllowTrailingCommas);
+            Assert.Equal(16 * 1024, options.DefaultBufferSize);
+            Assert.False(options.IgnoreNullValues);
+            Assert.Equal(0, options.MaxDepth);
+            Assert.Equal(JsonCommentHandling.Disallow, options.ReadCommentHandling);
+            Assert.False(options.WriteIndented);
+
+            // Setters should throw
```

Comment:
Even if being set to the same value? That's safe, no?

Reply:
I don't think that is valuable.

Reply:
The main scenario is getting the cursor in the property editor in a debugger.  If you hit escape, you cancel the edit.  If you hit enter, it assigns `foo.Prop = foo.Prop`, and there's a notice that an exception was thrown.

I've seen many people hit enter there and be frustrated by it.  So it's just a user-in-the-debugger niceness feature.  (I don't know if we have an official, or unofficial-but-de-facto stance).

Reply:
Hmm... I see. Where else have you seen it "fixed"?

Reply:
It's similar to https://source.dot.net/#System.Data.SqlClient/System/Data/SqlClient/SqlParameterHelper.cs,22b38c39fabe89a2, (setting a property to itself doesn't raise the PropertyChanged() event).  The logic extended to here is that the PropertyChanged() event is what should throw that the object is frozen.

Or List\<T\>.Capacity avoiding work when told to be the same value: https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/List.cs,9808f1f5ef16c436.

(Examples plucked from `findstr /sp /c:"(value != _" *.cs` and `findstr /sp /c:"!= value)" *.cs`.)

Reply:
In our case there is not any "extra" work. In fact one could argue for a tiny perf hit for comparing the value.

I'm not sold on comparing, but not necessarily against it either.